app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/UndertakingController.scala

```diff
@@ -196,6 +202,8 @@ class UndertakingController @Inject() (
         ref,
         None
       )
+      auditEventCreateUndertaking = createUndertakingAuditEvent(ref, undertaking)
```

Comment:
This conversion could be handled by the AuditService too. I dont' think it belongs in the controller. The service could provide a single method `auditCreateUndertaking(ref, undertaking)` that takes care of creating the audit event and logging it. `sendEvent` could then become a private method on the service.

Reply:
it belongs to controller because the values for creating that event, are in the controller like undertaking ref, response from creating an undertaking. This will be different for all 11 audit events

Reply:
for example the audit event for deleting a business entity and it's response can be get from business entity controller

Reply:
updated